Character.cpp

```diff
@@ -87,31 +91,61 @@ int Character::takeDamage(int damage)
 }
 
 
-#include <assert>
 void Character::attackInternal(Character& other)
 {
+    double statBoost = 1.1;
     if( other.hitPoints <= 0 )
     {
         /*
         When you defeat another Character: 
-            a) your stats are restored to their initial value
+            a) your stats are restored to their initial value if they are lower than it.
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+        if ( hitPoints < *initialHitPoints )
+        {
+            hitPoints = *initialHitPoints;
+        }
+
+        if ( armor < *initialArmorLevel )
+        {
+            armor = *initialArmorLevel;
+        }
+        
+        if ( attackDamage < *initialAttackDamage )
+        {
+            attackDamage = *initialAttackDamage;
+        }
+
+        *initialHitPoints = (hitPoints *= statBoost);
+        *initialArmorLevel = (armor *= statBoost);
+        *initialAttackDamage = (attackDamage *= statBoost);
```

Comment:
Your current stats aren't boosted, but your initial stats are.
- [ ] FIXME follow the Attack spec.